tests/voq/test_fabric_reach.py

```diff
@@ -0,0 +1,157 @@
+import pytest
+from tests.common.helpers.assertions import pytest_assert
+import logging
+import yaml
+
+logger = logging.getLogger(__name__)
+# This test only runs on t2 systems.
+pytestmark = [
+    pytest.mark.topology('t2')
+]
+
+supReferenceData = {}
+localModule = 0
+supervisorAsicBase = 300
+
+# Added a function to setup the reference data for sup.
+
+
+def test_setup_reference_data(duthosts):
+    # supReferenceData has the expected data for sup
+    global supReferenceData
+    keys = []
+    if len(duthosts.supervisor_nodes) == 0:
+        logger.info("Please run the test on modular systems")
+        return
+    duthost = duthosts.supervisor_nodes[0]
+    logger.info("duthost: {}".format(duthost.hostname))
+    num_asics = duthost.num_asics()
+    logger.info("num_asics: {}".format(num_asics))
+    for asic in range(num_asics):
+        keys.append('asic' + str(asic))
+    supReferenceData = {key: {} for key in keys}
+
+# This test checks the output of the "show fabric reachability" command
+# on one linecard. It is called once for each linecard in the chassis.
+# It loads the reference data for the linecard, runs the CLI command,
+# and compares the output.
+
+
+def test_fabric_reach_linecards(duthosts, enum_frontend_dut_hostname):
+    """compare the CLI output with the reference data"""
+    global supReferenceData
+    global localModule
+    global supervisorAsicBase
+
+    # supReferenceData has the expected data
+    if len(duthosts.supervisor_nodes) == 0:
+        logger.info("Please run the test on modular systems")
+        return
+    duthost = duthosts.supervisor_nodes[0]
+    logger.info("duthost: {}".format(duthost.hostname))
+
+    # Get hwSku for Fabriccards from the supervisor.
+    fabric_sku = None
+    fabric_sku = duthost.facts['hwsku']
+    pytest_assert(fabric_sku, "Need to add hwSku information for sup")
+
+    # Load the reference data file.
+    duthost = duthosts[enum_frontend_dut_hostname]
+    logger.info("duthost: {}".format(duthost.hostname))
+    lc_sku = duthost.facts['hwsku']
+    slot = duthost.facts['slot_num']
```

Comment:
In duthost.facts, we don't see 'slot_num' - is there some missing code that needs to be added to this PR in common/devices/sonic.py

Reply:
The slot_num variable is set in the chassis_facts function of the tests/voq/conftest.py file.
https://github.com/sonic-net/sonic-mgmt/blob/master/tests/voq/conftest.py#L25

I found that the slot_num variable is already present in the code when I checked it.

Reply:
Thanks for the clarification.